### PR TITLE
feat(News Article Schema): support enableNewsArticle prop from config

### PR DESCRIPTION
**Pull request by ReenaSingh07**
# Description

Created a new key in config to support the PB based toggle for News Article Schema

Fixes https://github.com/quintype/page-builder/issues/1459

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test Case A
- [ ] Test Case B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/generate-common-seo.js [hunk 1]
```diff
@@ -48,10 +48,13 @@ export function generateStructuredData(config = {}) {
   const { "theme-attributes":themeConfig, "social-links":socialLinks, "seo-metadata":seoMetadata = [] } = config;
   const homePageSeo = seoMetadata.find(page => page["owner-type"] === "home") || {};
   const { "page-title":pageTitle = "", description = "", keywords  =  "" } = get(homePageSeo, ["data"], {});
-
+  let enableNewsArticle = themeConfig['structured_data_news_article'];
   if(!themeConfig || !themeConfig.logo) {
     return {};
   }
+  if(config.hasOwnProperty('enableNewsArticle')){
```
Comment (EnKrypt) [inline src/generate-common-seo.js:55]: `hasOwnProperty` will return true if the value exists, even if it exists as `undefined`. Better to use the simpler syntax.

```suggestion
  if(config.enableNewsArticle){
```

File: src/generate-common-seo.js [hunk 2]
```diff
@@ -48,10 +48,13 @@ export function generateStructuredData(config = {}) {
   const { "theme-attributes":themeConfig, "social-links":socialLinks, "seo-metadata":seoMetadata = [] } = config;
   const homePageSeo = seoMetadata.find(page => page["owner-type"] === "home") || {};
   const { "page-title":pageTitle = "", description = "", keywords  =  "" } = get(homePageSeo, ["data"], {});
-
+  let enableNewsArticle = themeConfig['structured_data_news_article'];
```
Comment (EnKrypt) [inline src/generate-common-seo.js:51]: Variable name as it is does not convey adequate information, can be more verbose.

```suggestion
  let enableStructuredDataForNewsArticle = themeConfig['structured_data_news_article'];
```

File: src/generate-common-seo.js (hunk 1 above)
Comment (EnKrypt) [reply src/generate-common-seo.js:55]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#note

Comment (EnKrypt) [reply src/generate-common-seo.js:55]: My bad, this won't work because we need the condition to pass if the value is `false` as well.

There are a lot of hacks before and after this flow for that value, so for now `hasOwnProperty` along with a check for undefined should be fine.

File: src/generate-common-seo.js [hunk 3]
```diff
@@ -48,10 +48,13 @@ export function generateStructuredData(config = {}) {
   const { "theme-attributes":themeConfig, "social-links":socialLinks, "seo-metadata":seoMetadata = [] } = config;
   const homePageSeo = seoMetadata.find(page => page["owner-type"] === "home") || {};
   const { "page-title":pageTitle = "", description = "", keywords  =  "" } = get(homePageSeo, ["data"], {});
-
+  let enableStructuredDataForNewsArticle = themeConfig['structured_data_news_article'];
   if(!themeConfig || !themeConfig.logo) {
     return {};
   }
+  if(config.hasOwnProperty('enableNewsArticle') && typeof config.enableNewsArticle !== "undefined"){
+    enableStructuredDataForNewsArticle = config.enableNewsArticle;
```
Comment (EnKrypt) [inline src/generate-common-seo.js:56]:
```suggestion
    enableStructuredDataForNewsArticle = config.enableStructuredDataForNewsArticle ;
```

**Review by EnKrypt — APPROVED** [review]

**Review by atigrepravin — APPROVED** [review]

**Review by EnKrypt — APPROVED** [review]

**Review by Athira001 — APPROVED** [review]

File: src/generate-common-seo.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
-import omitBy from 'lodash/omitBy';
-import isUndefined from 'lodash/isUndefined';
 import get from 'lodash/get';
+import isUndefined from 'lodash/isUndefined';
```
Comment (Harshith46) [inline src/generate-common-seo.js:2]: why is this required

File: src/generate-common-seo.js [hunk 5]
```diff
@@ -48,10 +48,13 @@ export function generateStructuredData(config = {}) {
   const { "theme-attributes":themeConfig, "social-links":socialLinks, "seo-metadata":seoMetadata = [] } = config;
   const homePageSeo = seoMetadata.find(page => page["owner-type"] === "home") || {};
   const { "page-title":pageTitle = "", description = "", keywords  =  "" } = get(homePageSeo, ["data"], {});
-
   if(!themeConfig || !themeConfig.logo) {
     return {};
   }
+  let enableStructuredDataForNewsArticle = themeConfig['structured_data_news_article'] || false;
+  if(config.hasOwnProperty('enableStructuredDataForNewsArticle') && typeof config.enableStructuredDataForNewsArticle !== "undefined"){
```
Comment (Harshith46) [inline src/generate-common-seo.js:55]: `config.hasOwnProperty('enableStructuredDataForNewsArticle') && config.enableStructuredDataForNewsArticle ` should work

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@quintype/seo",
-  "version": "1.38.40",
+  "version": "1.38.41-news-article-schema.1",
```
Comment (Harshith46) [inline package.json:3]: check your package lock, try to keep only the version change in lock file

**Review by Harshith46 — CHANGES_REQUESTED** [review]

File: src/generate-common-seo.js (hunk 5 above)
Comment (ReenaSingh07) [reply src/generate-common-seo.js:55]: we want to check **config.enableStructuredDataForNewsArticle !== "undefined"** because **config.enableStructuredDataForNewsArticle** can also be false

Comment (ReenaSingh07) [reply src/generate-common-seo.js:55]: as per your condition only if it's true it will execute the if block

**Review by Harshith46 — APPROVED** [review]
